build-packages/merge-changelogs/index.ts

```diff
@@ -164,47 +142,40 @@ function mergeMessages(parsedMessages: Change[]): Change[] {
   }, [] as Change[]);
 }
 
-async function formatChangelog(parsedChangelogs: Change[]): Promise<string> {
-  const unifiedChangelog = await readFile('CHANGELOG.md', { encoding: 'utf8' });
-  const missingFromUnifiedChangelog = parsedChangelogs.filter(
-    summary => !unifiedChangelog.includes(`# ${summary.version}`)
+async function formatChangelog(messages: Change[]): Promise<string> {
+  if (!messages.length) {
+    throw new Error('No messages found in changelogs');
+  }
+
+  return (
+    formatMessagesOfType(messages, 'Known Issue') +
+    formatMessagesOfType(messages, 'Compatibility Note') +
+    formatMessagesOfType(messages, 'New Functionality') +
+    formatMessagesOfType(messages, 'Improvement') +
+    formatMessagesOfType(messages, 'Fixed Issue') +
+    formatMessagesOfType(messages, 'Updated Dependencies') +
+    '\n\n'
   );
-  const versions = [
-    ...new Set(missingFromUnifiedChangelog.map(msg => msg.version))
-  ];
-  return versions.reduce((changelog, version) => {
-    const newSection = createNewSection(
-      version,
-      missingFromUnifiedChangelog.filter(msg => msg.version === version)
-    );
-    return (
-      changelog.split('\n').slice(0, 30).join('\n') +
-      newSection +
-      changelog.split('\n').slice(30).join('\n')
-    );
-  }, unifiedChangelog);
 }
 
 export async function mergeChangelogs(): Promise<void> {
-  const workspaces: string[] = JSON.parse(
-    await readFile('package.json', { encoding: 'utf8' })
-  ).workspaces;
-  const workspacesWithVisibility = await Promise.all(
-    workspaces.map(async workspace => {
-      const packageJson = await readFile(resolve(workspace, 'package.json'), {
-        encoding: 'utf8'
-      });
-      return [!JSON.parse(packageJson).private, workspace] as const;
-    })
-  );
-  const pathsToPublicLogs = workspacesWithVisibility
-    .filter(([isPublic]) => isPublic)
-    .map(([, workspace]) => resolve(workspace, 'CHANGELOG.md'));
+  const { packages } = await getPackages(process.cwd());
```

Comment:
Nice replacement, out of curiosity, what made you look for an alternative?
/ What should trigger my search for replacements like this one in the future 🤔

Reply:
I needed this information independently of the package manager (as we want to use this for the other SDK).